src/pkg/packager/create.go

```diff
@@ -738,3 +796,56 @@ func (p *Packager) removeCopiesFromDifferentialPackage() error {
 
 	return nil
 }
+
+// addVexFile will perform vex schema validation and add the given file into the given path
+func addVexFile(componentPath string, path string, reportName string) error {
+	message.Debugf("path is: %s", path)
+	filePath, _ := os.Stat(path)
+	message.Debugf("filepath is: %s", filePath)
+	if !filePath.IsDir() {
+		message.Debugf("Vex path is a single file: %s", path)
+		dest := fmt.Sprintf("%s/%s/%s", componentPath, "vex", reportName)
+		if _, err := os.Stat(dest); err == nil {
+			return fmt.Errorf("%s already exists for this component and cannot conflict", dest)
+		}
+
+		doc, err := vex.Load(path)
+		if err != nil {
+			return fmt.Errorf("unable to load vex document for %s from %s: %w", reportName, path, err)
+		}
+		// Convert to JSON
+		var b bytes.Buffer
+		if err = doc.ToJSON(&b); err != nil {
+			return fmt.Errorf("unable to write vex doc for %s to JSON: %w", reportName, err)
+		}
+		message.Debugf("Loaded VEX file for %s (%d bytes)", reportName, b.Len())
+
+		// Write VEX file to the vex directory
+		if err = utils.WriteFile(dest, b.Bytes()); err != nil {
+			return fmt.Errorf("unable to write vex file to %s: %w", dest, err)
+		}
+	} else {
+		message.Debugf("Vex path is a directory!")
+		file, err := os.Open(path)
+
+		if err != nil {
+			return fmt.Errorf(lang.PkgValidateErrPath, err)
+		}
+
+		defer file.Close()
+
+		files, err := file.Readdirnames(0)
+		if err != nil {
+			return fmt.Errorf(lang.PkgValidateErrPath, err)
+		}
+
+		message.Debugf("VEX files are: %s", files)
+
+		for i, f := range files {
+			filename := fmt.Sprintf("%s-%d.vex.json", reportName, i)
+			filepath := fmt.Sprintf("%s/%s", path, f)
+			addVexFile(componentPath, filepath, filename)
+		}
+	}
+	return nil
+}
```

Comment:
I think this is worthy of creating a new dir `src/pkg/reports` creating a `vex.go` there